src/App/Property.cpp

```diff
@@ -48,11 +50,12 @@ TYPESYSTEM_SOURCE_ABSTRACT(App::Property , Base::Persistence)
 //**************************************************************************
 // Construction/Destruction
 
+static std::atomic<long> _PropID;

```

Comment:
Could we define this to be an `uint64_t` instead?
On some configurations `long` can be 32bits.
If someone would use such a system and this static variable would get updated 60 times a second it would only take about a year before it wraps around and becomes negative. Give it another year and the uniqueness aren't guaranteed anymore.

I do know that this is an edge case, but better safe than sorry when it comes to uniqueness. Edge cases has happened before :)